code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -1701,6 +1704,104 @@ func (srh *StorageRestHandler) getAllocationMinLock(w http.ResponseWriter, r *ht
 	}, nil)
 }
 
+// swagger:route GET /v1/screst/6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7/allocation-update-min-lock allocation-update-min-lock
+// Calculates the cost for updating an allocation.
+//
+// parameters:
+//
+//	+name: data
+//	 description: json marshall of update allocation request input data
+//	 in: query
+//	 type: string
+//	 required: true
+//
+// responses:
+//
+//	200: Int64Map
+//	400:
+//	500:
+func (srh *StorageRestHandler) getAllocationUpdateMinLock(w http.ResponseWriter, r *http.Request) {
+	balances := srh.GetStateContextI()
+	edb := balances.GetEventDB()
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
+		return
+	}
+
+	data := r.URL.Query().Get("data")
+	var req updateAllocationRequest
+	if err := req.decode([]byte(data)); err != nil {
+		common.Respond(w, r, "", common.NewErrInternal("can't decode allocation request", err.Error()))
+		return
+	}
+
+	eAlloc, err := edb.GetAllocation(req.ID)
+	if err != nil {
+		common.Respond(w, r, nil, common.NewErrBadRequest(err.Error()))
+		return
+	}
+
+	alloc, err := allocationTableToStorageAllocationBlobbers(eAlloc, edb)
+	if err != nil {
+		common.Respond(w, r, nil, common.NewErrInternal(err.Error()))
+		return
+	}
+
+	conf, err := getConfig(balances)
+	if err != nil {
+		common.Respond(w, r, nil, common.NewErrInternal(err.Error()))
+		return
+	}
+
+	sc := StorageSmartContract{
+		SmartContract: &sci.SmartContract{
+			ID: ADDRESS,
+		},
+	}
+
+	txn := &transaction.Transaction{
+		CreationDate: common.Now(),
+	}
+
+	blobbers := make([]*StorageNode, 0, len(alloc.Blobbers))
+	for _, br := range alloc.Blobbers {
+		b := StoragNodeResponseToStorageNode(*br)
+		blobbers = append(blobbers, &b)
+	}
+
+	_, err = sc.updateAllocationWithOwner(balances, txn, &req, conf, &alloc.StorageAllocation, blobbers)
```

Comment:
we can't run any MPT modification operations from rest endpoints, even fake ones, that's why QueryChainer doesn't provide real StateContext. Naming is not correct, since we doesn't do any update there, but even selects should be done from events db, but not from MPT.
Can we do `sa.cost()` and `sa.restMinLockDemand()` using eventsdb instead?

Reply:
Yes, I agree we should use the eventDB when possible. I did this for simplicity. I tried to do it with event db only, but the update action is a bit complicate than the new allocation as we need to calculate the average terms to get new restMinLockDemand. Let me check again.